packages/flutter/lib/src/widgets/image.dart

```diff
@@ -91,7 +115,11 @@ Future<void> precacheImage(
       if (!completer.isCompleted) {
         completer.complete();
       }
-      stream.removeListener(listener);
+      // Give callers until at least the end of the frame to subscribe to the
+      // image stream.
```

Comment:
`See also: [ImageCache._liveImages]`?

Reply:
That could make sense here, sure.  I'll add that.